xbmc/PlayListPlayer.cpp

```diff
@@ -187,10 +185,10 @@ bool CPlayListPlayer::PlayNext(int offset, bool bAutoPlay)
     if(!bAutoPlay)
       CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Info, g_localizeStrings.Get(559), g_localizeStrings.Get(34201));
 
-    CGUIMessage msg(GUI_MSG_PLAYLISTPLAYER_STOPPED, 0, 0, m_iCurrentPlayList, m_iCurrentSong);
+    CGUIMessage msg(GUI_MSG_PLAYLISTPLAYER_STOPPED, 0, 0, static_cast<int>(m_iCurrentPlayList), m_iCurrentSong);
```

Comment:
Seeing a lot of `CGUIMessage`constructor calls passing a static_casted PlayListId as parameter. maybe you should add another `CGUIMessage`contructor, directly accepting a parameter of type `PlaylistId`? Implementation could do the static_cast, advantage is that you have this case only at one place.